contrib/babelfishpg_tsql/src/pltsql_utils.c

```diff
@@ -2135,13 +2142,30 @@ exec_alter_role_cmd(char *query_str, RoleSpec *role)
  * Helper function to generate GRANT on SCHEMA subcommands.
  */
 static List
-*gen_grantschema_subcmds(const char *schema, const char *rolname, bool is_grant, bool with_grant_option, AclMode privilege)
+*gen_grantschema_subcmds(const char *schema, const char *rolname, bool is_grant, bool with_grant_option, AclMode privilege, bool is_create_schema)
 {
 	StringInfoData query;
 	List	   *stmt_list;
 	Node	   *stmt;
 	int			expected_stmts = 2;
 	int			i = 0;
+	bool		owner_other_than_dbo = false;
+	char 		*dbname = get_cur_db_name();
+	const char	*dbo_role = get_dbo_role_name(dbname);
+	const char	*db_owner_role = get_db_owner_name(dbname);
+	const char	*schema_owner = GetUserNameFromId(get_owner_of_schema(schema), false);
+
+	/*
+	 * Don't need multiple ALTER DEFAULT PRIVILEGE statements, if:
+	 * 1. It's a part of CREATE SCHEMA statement
```

Comment:
Can you please explain the 1st point? Which grant statement gets executed at the time of grant schema? Is it `grant USAGE on schema to public` ?

Reply:
As a part of `CREATE SCHEMA .. AUTHORIZATION ..`, we call `exec_grantschema_subcmds`. We don't need to execute these `ALTER DEFAULT PRIVILEGES` on multiple roles as a part of `CREATE SCHEMA` statement.